microsoft/botframework-solutions
# [Enterprise Bot Template][TypeScript] Fix TSLint Warnings

## Description
- Change tslint severity to **error**.
- Add tslint's exceptions.
- Fix the TSLint warnings.

## Related Issue
Fix #679 

## Additional context
**Once the TSLint severity is set to "Error", any change made to the sample should follow the TSLint rules; exceptions should be avoided when possible, and if not, they should be properly justified**

## Testing Steps
1. Open `AI/templates/Enterprise-Template/src/typescript/enterprise-bot/`
2. Run `npm i`.
3. Run `npm run lint` and check there is no tslint errors/warnings in the solution.

## Checklist
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have added or updated the appropriate unit tests~
~- [ ] I have tested any new/updated dialogs using Speech in the emulator to ensure the [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) property is set to enable a high quality speech-first experience and the appropriate [InputHints](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) are set correctly.~ 
~- [ ] I have updated related documentation~

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
~- [ ] A duplicate issue is filed to track future  work.~

If you have updated responses or `.lu` files:
~- [ ] All languages have been updated~
~- [ ] You have tested deployment with your new models~
